dist/chart-legend/chart-legend.css

```diff
@@ -0,0 +1,103 @@
+.chart-legend {
+  display: inline-block;
+}
+.chart-legend [hidden] {
+  border: 0;
+  clip: rect(0, 0, 0, 0);
+  height: 1px;
+  margin: -1px;
+  overflow: hidden;
+  padding: 0;
+  position: absolute;
+  white-space: nowrap;
+  width: 1px;
+}
+.chart-legend.table {
+  flex-shrink: 0;
+}
+.chart-legend ul {
+  list-style: none;
+  margin: 0;
+  -webkit-padding-start: 24px;
+          padding-inline-start: 24px;
+}
+.chart-legend ul li {
+  line-height: 2rem;
+  position: relative;
+  transition: opacity 0.5s;
+}
+.chart-legend ul li.primary::before {
+  background-color: var(--data-viz-primary, var(--color-data-viz-primary));
+}
+.chart-legend ul li.secondary::before {
+  background-color: var(--data-viz-secondary, var(--color-data-viz-secondary));
+}
+.chart-legend ul li.tertiary::before {
+  background-color: var(--data-viz-tertiary, var(--color-data-viz-tertiary));
+}
+.chart-legend ul li.quaternary::before {
+  background-color: var(--data-viz-quaternary, var(--color-data-viz-quaternary));
+}
+.chart-legend ul li.quinary::before {
+  background-color: var(--data-viz-quinary, var(--color-data-viz-quinary));
+}
+.chart-legend ul li.senary::before {
+  background-color: var(--data-viz-senary, var(--color-data-viz-senary));
+}
+.chart-legend ul li.septenary::before {
+  background-color: var(--data-viz-septenary, var(--color-data-viz-septenary));
+}
+.chart-legend ul li.octanary::before {
+  background-color: var(--data-viz-octanary, var(--color-data-viz-octanary));
+}
+.chart-legend ul li.pattern-0::before {
+  background: repeating-linear-gradient(-45deg, var(--data-viz-tertiary, var(--color-data-viz-tertiary)), var(--data-viz-tertiary, var(--color-data-viz-tertiary)) 2px, var(--data-viz-primary, var(--color-data-viz-primary)) 2px, var(--data-viz-primary, var(--color-data-viz-primary)) 4px);
+  background-color: var(--data-viz-primary, var(--color-data-viz-primary));
+}
+.chart-legend ul li.pattern-1::before {
+  background-color: var(--data-viz-septenary, var(--color-data-viz-septenary));
+  background-image: repeating-linear-gradient(45deg, var(--data-viz-senary, var(--color-data-viz-senary)) 25%, transparent 25%, transparent 75%, var(--data-viz-senary, var(--color-data-viz-senary)) 75%, var(--data-viz-senary, var(--color-data-viz-senary))), repeating-linear-gradient(45deg, var(--data-viz-senary, var(--color-data-viz-senary)) 25%, var(--data-viz-septenary, var(--color-data-viz-septenary)) 25%, var(--data-viz-septenary, var(--color-data-viz-septenary)) 75%, var(--data-viz-senary, var(--color-data-viz-senary)) 75%, var(--data-viz-senary, var(--color-data-viz-senary)));
+  background-position: 0 0, 6px 6px;
+  background-size: 12px 12px;
+}
+.chart-legend ul li.pattern-2::before {
+  background-color: var(--data-viz-quinary, var(--color-data-viz-quinary));
+}
+.chart-legend.table ul {
```

Comment:
Similar to the comment above about `primary` and `secondary`, having a classname of `.table` we risk other page CSS cascading and leaking into our component if the page has defined a `.table` class. You might want to consider something like `chart-legend__table`.